ingest/src/sources/mongodb-university/MongoDbUniversityDataSource.ts

```diff
@@ -0,0 +1,93 @@
+import { Page, PageMetadata } from "mongodb-rag-core";
+import { DataSource } from "../DataSource";
+import { makeUniversityPages } from "./makeUniversityPages";
+import {
+  TiCatalogItem,
+  makeMongoDbUniversityDataApiClient,
+} from "./MongoDbUniversityDataApiClient";
+
+/**
+  Parameters for constructing a MongoDB University Data API source.
+ */
+export interface MakeMongoDbUniversityDataSourceParams {
+  /**
+      Name of the MongoDB University Data API source.
+      Maps to the {@link DataSource.name}.
+     */
+  sourceName: string;
+
+  /**
+      Base URL for the MongoDB University Data API.
+     */
+  baseUrl: string;
+
+  /**
+      API key for the MongoDB University Data API.
+     */
+  apiKey: string;
+
+  /**
+      Filter function for filtering out items from the MongoDB University
+      catalog. For example, you may want to only ingest items that are
+      in public content.
+
+      To only ingest public, published, and non-legacy
+      MongoDB University content, use the
+      {@link filterOnlyPublicActiveTiCatalogItems} filter function.
+     */
+  tiCatalogFilterFunc: (item: TiCatalogItem) => boolean;
+
+  /**
+      Metadata for the MongoDB University Data API source.
+      Included in the {@link Page.metadata} for each page.
+     */
+  metadata?: PageMetadata;
+}
+
+/**
+  Filter function to only include public, published,
+  and non-legacy MongoDB University content.
+
+  > ⚠️ **Important** ⚠️
+  >
+  > You should include *only* this content or a subset of it
+  > in externally facing applications.
+ */
+export const filterOnlyPublicActiveTiCatalogItems: MakeMongoDbUniversityDataSourceParams["tiCatalogFilterFunc"] =
```

Comment:
@amanda/yalla: could you verify that this filter seems correct?

Reply:
`item.associated_videos.length > 0` will exclude lab only courses (eg: [Shell Scripting with mongosh and jq](https://learn.mongodb.com/courses/shell-scripting-with-mongosh-and-jq)), as well as higher level courses and learning paths that are made up of units, which have an `associated_content` field with units and learning bytes that have videos (eg: [Using MongoDB with Node.js](https://learn.mongodb.com/learning-paths/using-mongodb-with-nodejs)).

Reply:
>my thought was to exclude the lab only courses.
> if it has this filter, would it exclude any video content?

It shouldn't. If it was intended to only capture courses with videos, then this is fine.

Reply:
It will exclude higher level courses and learning paths which don't have videos directly on them, but are made up of units and learning bytes that do. I'm not sure if that is a problem for you or not, I guess it depends on how you intend to use it.

Reply:
> It will exclude higher level courses and learning paths which don't have videos directly on them, but are made up of units and learning bytes that do. I'm not sure if that is a problem for you or not, I guess it depends on how you intend to use it.

we are currently only using the video transcripts. though if the text content associated with these higher level courses and learning paths were included in the data API, it could make sense to modify this filter and ingest that content as well.

Reply:
> if you want the chatbot to recommend a course or learning path

i think the current behavior is desired b/c we don't have any additional data about the course/learning path (like a description).

if we had some more content associated with these catalog items, then it would make sense to include, but we don't currently. including that additional data in the API, and using it to ingest pages could make sense as a future addition to the data api and this data source.